client/components/parts/ImageSearchField.jsx

```diff
@@ -28,63 +27,64 @@ export default class ImageSearchField extends Component {
+
+  searchByKeyword = () => new Promise((resolve, reject) => {
+    clearTimeout(this.searchTimer)
+    this.searchTimer = setTimeout(() => {
+      // Send search request
+      fetch(`https://api.unsplash.com/search/photos/?client_id=${Config.unsplash_client_id}&query=${this.state.searchKeywords}`)
+      .then((response) => {
+        if (response.ok){
+          resolve(response.json().then((json) => (json.results).map((imgObject) =>
+            ({
+              'thumbnail': imgObject.urls.thumb,
+              'full': imgObject.urls.full
+            })
+          )))
+        } else {
+          reject(response.status)
+        }
       })
-    }))
-  }
+    }, 500) // <- Wait 500ms before querying
+  })
 
   render() {
-    // ToDo: create a style file & port 'height', 'width', etc. attributes to it
     return (
-      <div className={classNames({'hidden': !this.props.isActive})} id="heroImageSearch" >
-        <label className={classNames('form-label')}>{this.props.label}</label>
-        <input onChange={this.updateKeywords}
-               maxLength="64"
-               className="form-input"
-               type="url"
-               value={this.state.searchRawString}
-               placeholder={this.props.placeholder} />
-        <div className={classNames('columns')} id="thumbnails">
-          <div className={classNames('column', 'col-md-3')}>
-            <Async
-              promise={this.searchByKeyword()}
-              then={(imgUrls) => {
-                return (
-                  <div>
-                    {
-                      imgUrls.map((urls, index) => {
-                        return (
-                          <img onClick={this.handleChange}
-                               className={classNames('thumbnail')}
-                               src={urls.thumbnail}
-                               id={urls.full}
-                               key={`img${index}`} />
-                         )
-                      })
-                     }
-                   </div>
-                 )
-              }}>
-            </Async>
-          </div>
+      <div className="form-image-search">
+        <input onChange={this.updateKeywords} maxLength="64" className="form-input" type="url" value={this.state.searchRawString} placeholder={this.props.placeholder} />
+        <div className="thumbnails">
+          <Async
+            promise={this.searchByKeyword()}
+            then={(imgUrls) =>
+              <div className="columns col-gapless">
+                {imgUrls.slice(0, 4).map((urls, index) =>
+                  <div className="column col-6 thumbnail" key={`thumbnail-${index}`}>
+                    <button type="button" className="btn btn-link btn-thumbnail" onClick={this.handleChange}>
+                      <img className="img-responsive rounded" src={urls.thumbnail} data-url={urls.full} />
+                    </button>
+                  </div>
+                  )}
+              </div>
```

Comment:
I think in the future it'd be better practice to write functions that output html above the return statement in the `render()` function. This is fine for now though